.github/workflows/callable-test.yaml

```diff
@@ -37,18 +37,87 @@ on:
         # renovate: datasource=github-tags depName=defenseunicorns/uds-cli versioning=semver
         default: 0.27.8
         type: string
+      file-filter:
+        type: string
+        description: |
+          A list of patterns to filter files that should not trigger package testing.
+          Entries use bash glob matching, patterns are separated by newlines.
+        default: |
+          **.md
+          **.jpg
+          **.png
+          **.gif
+          **.svg
+          adr/**
+          docs/**
+          .gitignore
+          renovate.json
+          .release-please-manifest.json
+          release-please-config.json
+          oscal-component.yaml
+          CODEOWNERS
+          LICENSE
+          CONTRIBUTING.md
+          SECURITY.md
+          config/renovate.json5
 
 # Permissions for the GITHUB_TOKEN used by the workflow.
 permissions:
   contents: read # Allows reading the content of the repository.
   packages: read # Allows reading the content of the repository's packages.
   id-token: write
+  pull-requests: read
 
 jobs:
+  doc-shim-check:
+    runs-on: ubuntu-latest
+    outputs:
+      test: ${{ steps.check-files.outputs.test }}
+    steps:
+      - name: Check for non-documentation changes
+        id: check-files
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          FILE_FILTER_INPUT: ${{ inputs.file-filter }}
+        run: |
+          # allow ** for recursive globs
+          shopt -s globstar
+
+          MODIFIED_FILES=$(gh pr view ${{ github.event.pull_request.number }} \
+            --repo ${{ github.repository }} --json files \
+            --jq '.files[].path')
```

Comment:
This is probably fine since we run CI on each commit, but there might be some timing here where `gh pr view` gets a different set of files/diffs than the commit that actually got pushed (this is an API call rather than looking at the current commit diffs).

Alternative here might be something like `git diff --name-only ${{ github.event.pull_request.head.sha }}` (I think that's the right var), which would get the list of files changed from the current commit back to the head of the PR.

Reply:
The timing I'm thinking of is actually two commits in quick succession. The first commit might get the diffs from the second commit, because the PR has new commits. Again probably irrelevant since we cancel CI when newer commits happen. 

For what its worth dorny/paths-filter uses `git` commands only I think ([ref](https://github.com/dorny/paths-filter/blob/master/src/git.ts)). I don't think it's using the gh CLI/API?